src/redux/books/books.js

```diff
@@ -1,28 +1,67 @@
-const ADD_BOOK = 'bookStore/books/addBook';
-const REMOVE_BOOK = 'bookStore/books/removeBook';
+import { v4 as uuidv4 } from 'uuid';
 
-const initialState = {
-  books: [],
-};
+const ADD_BOOK = 'bookStore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
 
-export function addBook(books) {
+const initialState = [{
+  id: uuidv4(),
+  title: 'The Hunger Games',
+  author: 'Suzanne Collins',
+  category: 'Action',
+  progress: 64,
+  currentChapter: 17,
+},
+{
+  id: uuidv4(),
+  title: 'Dune',
+  author: 'Frank Herbert',
+  category: 'Science Fiction',
+  progress: 8,
+  currentChapter: 3,
+},
+{
+  id: uuidv4(),
+  title: 'Capital in the Twenty-First century',
+  author: 'Suzanne Collins',
+  category: 'Economy',
+  progress: 0,
+  currentChapter: 'introduction',
+},
+];
+
+export function addBook(newBook) {
   return {
     type: ADD_BOOK,
-    books,
+    payload: {
+      id: uuidv4(),
+      title: newBook.title,
+      author: newBook.author,
+      category: newBook.category,
+      progress: 0,
+      currentChapter: 0,
+    },
   };
 }
 
-export function removeBook(books) {
+export function removeBook(id) {
   return {
     type: REMOVE_BOOK,
-    books,
+    payload: {
+      bookid: id,
+    },
   };
 }
 
 function reducer(state = initialState, action) {
   switch (action.type) {
     case ADD_BOOK:
-    default: return state.books;
+      return [...state, action.payload];
+
+    case REMOVE_BOOK:
+      return [state.filter((book) => book.id !== action.payload.id)];
+
```

Comment:
- [ ] Kindly check remove k=logic for removing book, there is a strange behaviour when removing it, you can check [this video](https://www.loom.com/share/0ace5020c928452bb561896ef1e6abd4) for what I mean.